src/otx/algo/samplers/class_incremental_sampler.py

```diff
@@ -0,0 +1,179 @@
+
+        if not len(self.new_indices) > 0:
+            self.new_indices = self.old_indices
+            self.old_indices = torch.tensor([], dtype=torch.int64)
+
+        old_new_ratio = np.sqrt(len(self.old_indices) / len(self.new_indices))
+
+        if efficient_mode:
+            self.data_length = int(len(self.new_indices) * (1 + old_new_ratio))
+            self.old_new_ratio = 1
+        else:
+            self.data_length = len(self.dataset)
+            self.old_new_ratio = int(old_new_ratio)
+
+        self.num_samples = self._calcuate_num_samples()
+
+    def _calcuate_num_samples(self) -> int:
+        num_samples = self.repeat * (1 + self.old_new_ratio) * int(self.data_length / (1 + self.old_new_ratio))
+
+        if not self.drop_last:
+            num_samples += (
+                int(np.ceil(self.data_length * self.repeat / self.batch_size)) * self.batch_size - num_samples
+            )
+
+        if self.num_replicas > 1:
+            # If the dataset length is evenly divisible by # of replicas, then there
+            # is no need to drop any data, since the dataset will be split equally.
+            if self.drop_last and num_samples % self.num_replicas != 0:
+                # Split to nearest available length that is evenly divisible.
+                # This is to ensure each rank receives the same amount of data when
+                # using this Sampler.
+                num_samples = math.ceil(
+                    # `type:ignore` is required because Dataset cannot provide a default __len__
+                    # see NOTE in pytorch/torch/utils/data/sampler.py
+                    (num_samples - self.num_replicas) / self.num_replicas,
+                )
+            else:
+                num_samples = math.ceil(num_samples / self.num_replicas)
+            self.total_size = num_samples * self.num_replicas
+
+        return num_samples
+
+    def __iter__(self):
+        """Iter."""
+        if self.generator is None:
+            seed = int(torch.empty((), dtype=torch.int64).random_().item())
+            generator = torch.Generator()
+            generator.manual_seed(seed)
+        else:
+            generator = self.generator
+
+        indices = []
+        for _ in range(self.repeat):
+            for _ in range(int(self.data_length / (1 + self.old_new_ratio))):
+                indice = torch.cat(
+                    [
+                        self.new_indices[torch.randint(0, len(self.new_indices), (1,), generator=generator)],
+                        # random_generator.choice(self.new_indices, 1),
+                        self.old_indices[
+                            torch.randint(0, len(self.old_indices), (self.old_new_ratio,), generator=generator)
+                        ],
+                        # random_generator.choice(self.old_indices, self.old_new_ratio),
+                    ],
+                )
+                indices.append(indice)
```

Comment:
```suggestion
        for _ in range(self.repeat):
            num_samples = int(self.data_length / (1 + self.old_new_ratio))
            indice = torch.cat(
                    [
                        self.new_indices[torch.randint(0, len(self.new_indices), (num_samples,), generator=generator)],
                        self.old_indices[
                            torch.randint(0, len(self.old_indices), (num_samples * self.old_new_ratio,), generator=generator)
                        ],
                    ],
                )
```

Reply:
Thanks for the comment. Modify this to create indices separated by batches. - https://github.com/openvinotoolkit/training_extensions/pull/3018/commits/6a1b4e730243addae624727759215803e0a62744